pkg/kv/kv.go

```diff
@@ -30,6 +30,95 @@ import (
 
 var extraHandleColumnInfo = model.NewExtraHandleColInfo()
 
+// PairIter abstract iterator method for Ingester.
+type PairIter interface {
+	// IsEmpty checks whether iter has no pairs.
+	IsEmpty() bool
+	// Error return current error on this iter.
+	Error() error
+	// First return the first key in this iter.
+	First() []byte
+	// Last return the last key in this iter.
+	Last() []byte
+	// Valid check this iter reach the end.
+	Valid() bool
+	// Next moves this iter forward.
+	Next() bool
+	// Key represents current position pair's key.
+	Key() []byte
+	// Value represents current position pair's Value.
+	Value() []byte
+	// Close close this iter.
+	Close() error
+}
+
+// SimplePairIter represents simple pair iterator.
+// which is used for log restore.
+type SimplePairIter struct {
+	index int
+	pairs Pairs
+}
+
+// NewSimpleKeyIter creates SimplePairIter.
+func NewSimpleKeyIter(pairs Pairs) PairIter {
+	return &SimplePairIter{
+		index: 0,
```

Comment:
With a fresh new iterator without calling `Next` for the first time, the iterator should be invalid then. And the first call `Next`'s return value should be the same as `First`

Reply:
Done